### ISSUE TITLE
 My tests don't work very well, am I missing something？  

### ISSUE THREAD

**Issue by zhanghongyong123456** [qiuyu96/CoDeF#6]
1. my  base_control  image ：
![00017-2710618556](https://github.com/qiuyu96/CoDeF/assets/48466610/aa187b5f-9fdd-4f15-a9ce-d082c599f5e6)
 my  transformed   video:

https://github.com/qiuyu96/CoDeF/assets/48466610/4aa9aa29-93c8-4b1a-9dae-7546e13dbeca

2. If  I train my own model（my video）, do I only need one image sequence?
how to train myself videos

3.  about  base_control （by ControlNet v1.1   ues which meath: depth?Openpose?SoftEdge or another)

**Comment by qiuyu96** [comment]
Thanks for your attention.
1. What's the `canonical_wh` in your setting? It should be larger than the `img_wh`. We recommend using our default setting. The canonical image of this sequence should be like this when using our default setting:
![canonical_0](https://github.com/qiuyu96/CoDeF/assets/44515536/5d2b22b1-57ef-44c5-a02c-c25c8c79ae42)
Then feed the **raw** canonical image to ControlNet to get the transferred canonical image.
It appears that the field of view of your transferred canonical image is smaller. Please review this carefully.

2. To train your own model, the most straightforward approach is to store your video sequences in the `all_sequences` folder. In certain cases, you may need to include optical flow and masks, along with meticulous tuning of hyperparameters in the configuration file. Make sure to organize the files in a similar manner as our exemplar data. 

3. The choice of the ControlNet model depends on the specific requirements of the situation. In this example, we utilize the [ControlNet 1.1 Anime Lineart model](https://github.com/lllyasviel/ControlNet-v1-1-nightly#controlnet-11-anime-lineart) to generate the desired transferred image.

**Comment by zhanghongyong123456** [comment]
> Thanks for your attention.
> 
> 1. What's the `canonical_wh` in your setting? It should be larger than the `img_wh`. We recommend using our default setting. The canonical image of this sequence should be like this when using our default setting:
>    ![canonical_0](https://user-images.githubusercontent.com/44515536/261552125-5d2b22b1-57ef-44c5-a02c-c25c8c79ae42.png)
>    Then feed the **raw** canonical image to ControlNet to get the transferred canonical image.
>    It appears that the field of view of your transferred canonical image is smaller. Please review this carefully.
> 2. To train your own model, the most straightforward approach is to store your video sequences in the `all_sequences` folder. In certain cases, you may need to include optical flow and masks, along with meticulous tuning of hyperparameters in the configuration file. Make sure to organize the files in a similar manner as our exemplar data.
> 3. The choice of the ControlNet model depends on the specific requirements of the situation. In this example, we utilize the [ControlNet 1.1 Anime Lineart model](https://github.com/lllyasviel/ControlNet-v1-1-nightly#controlnet-11-anime-lineart) to generate the desired transferred image.

1. I  test  beauty_0 file，All are the default settings, except for the addition of a folder（base_control） 512x512  base_control image，
![image](https://github.com/qiuyu96/CoDeF/assets/48466610/7094a252-48b8-4336-bec2-400e440b75a7)
2. https://github.com/qiuyu96/CoDeF#customize-your-own-data，  Will this 《To be released.》 the code of how to get the flow and the corresponding mask？

**Comment by qiuyu96** [comment]
1. How does your canonical image appear once you execute the command `./scripts/test_multi.sh`?

2. We obtain flow using [RAFT]() and mask using [SAM-Track](https://github.com/z-x-yang/Segment-and-Track-Anything).

**Comment by zhanghongyong123456** [comment]
> 1. How does your canonical image appear once you execute the command `./scripts/test_multi.sh`?
> 2. We obtain flow using RAFT and mask using [SAM-Track](https://github.com/z-x-yang/Segment-and-Track-Anything).

when   i  run this  cmd(./scripts/test_multi.sh), i will get canonical image
![image](https://github.com/qiuyu96/CoDeF/assets/48466610/46bd903e-23a8-45ff-9294-ea5962159242)
i run   this cmd(./scripts/test_canonical.sh),i get this file 
![image](https://github.com/qiuyu96/CoDeF/assets/48466610/752641fa-7d16-4a97-a671-8ddb872c8260)

**Comment by zhanghongyong123456** [comment]
> 1. How does your canonical image appear once you execute the command `./scripts/test_multi.sh`?
> 2. We obtain flow using RAFT and mask using [SAM-Track](https://github.com/z-x-yang/Segment-and-Track-Anything).

about  RAFT ,when    i  run   project's  demo.py ,
I can get the optical flow and the original image stitching image, so how to get the other files，like[  0000.npy、 matches.png、warped.png  ]in the beauty_flow folder, and [*_c.npy]  in the beauty_0_flow_confidence file
![image](https://github.com/qiuyu96/CoDeF/assets/48466610/92154ac2-cf2e-4518-925f-20f5770e9250)

**Comment by ken-ouyang** [comment]
Hi hongyong, 

The translated canonical image seems not align with the one you get from the ./scripts/test_multi.sh. We recommend you use controlNet 1.1 Lineart or LineartAnime models to achieve a better aligned canonical.  

For Raft, you can directly get the flow.npy from the original code. But for confidence, you may check the paper for the detailed calculation. We will improve and release the preprocess scripts in a short time. 

Regards,
Hao

**Comment by jfischoff** [comment]
@ken-ouyang, 

I have some questions about this section.
![Screenshot from 2023-08-21 13-31-58](https://github.com/qiuyu96/CoDeF/assets/168507/b051f120-cb2a-43b5-ad11-b675cd05dc2f)

1. What epsilon value should be used?
2. Is the confidence tensor either 1 or 0 thresholded by epsilon, or is the epsilon filtering the values, so that they are 0 if less than epsilon and unchanged if greater?

**Comment by ken-ouyang** [comment]
@jfischoff 
1.  Epsilon can be set to around 5 to 10(RGB range: 0~255). 
2. Yes the value is binary. But I think the soft mask will work as well. 
We are planning to release the preprocess scripts very soon. The detailed settings will be crystal clear. Please stay tuned.

**Comment by ken-ouyang** [comment]
@jfischoff It is released.

**Comment by frankchieng** [comment]
> Hi hongyong,
> 
> The translated canonical image seems not align with the one you get from the ./scripts/test_multi.sh. We recommend you use controlNet 1.1 Lineart or LineartAnime models to achieve a better aligned canonical.
> 
> For Raft, you can directly get the flow.npy from the original code. But for confidence, you may check the paper for the detailed calculation. We will improve and release the preprocess scripts in a short time.
> 
> Regards, Hao

i tinker with sdxl linearart controlnet of [t2i-adapter-lineart-sdxl-1.0](https://huggingface.co/TencentARC/t2i-adapter-lineart-sdxl-1.0) ,but when i applied controlnet and set the strength value of 1.0,the transferred image renders so ugly, when i decrease the strength value around 0.5,the final image quality is acceptable,but the motion and gesture of the profile changed,so how can i solve this problem then i can run test_cannonical.sh
